### ISSUE TITLE
CI: K8sServicesTest Checks service across nodes with L7 policy Tests NodePort with L7 Policy: tftp connect via IPv6 failed

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#10118]
```

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.17-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:430
Pod "testclient-4h27j" can not connect to service "tftp://[::ffff:192.168.36.11]:31270/hello"
Expected command: kubectl exec -n default testclient-4h27j -- curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 8 tftp://[::ffff:192.168.36.11]:31270/hello -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
To succeed, but it failed:
Exitcode: 28 
Stdout:
 	 time-> DNS: '0.000037()', Connect: '0.000074',Transfer '0.000000', total '4.999814'
Stderr:
 	 command terminated with exit code 28
	 

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.17-gopath/src/github.com/cilium/cilium/test/k8sT/Services.go:373
```

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/16973/testReport/junit/Suite-k8s-1/17/K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy/

**Comment by tgraf** [comment]
Hit in https://github.com/cilium/cilium/pull/10928

```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.18-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:430
Pod "testclient-ggfcl" can not connect to service "tftp://[::ffff:192.168.36.11]:32012/hello"
Expected command: kubectl exec -n default testclient-ggfcl -- curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 8 tftp://[::ffff:192.168.36.11]:32012/hello -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
To succeed, but it failed:
Exitcode: 72 
Stdout:
 	 time-> DNS: '0.000016()', Connect: '0.000031',Transfer '0.000000', total '0.003627'
Stderr:
 	 command terminated with exit code 72
	 

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.18-gopath/src/github.com/cilium/cilium/test/k8sT/Services.go:618
```

[754f56b5_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip](https://github.com/cilium/cilium/files/4464279/754f56b5_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip)

**Comment by jrajahalme** [comment]
This test is currently disabled due to a consistent failure in GKE/Azure/ENI (#11235).

**Comment by jrajahalme** [comment]
Note that there is no proxy redirection for TFTP, so this failure should have nothing to do with L7/proxy redirection.

**Comment by jrajahalme** [comment]
The failing test traffic is seemingly between nodes (source POD and 192.168.36.11 may be in different nodes), but the host port service load balancing at the destination can pick a backend on either node. It is likely that one of these scenarios (backend in the same node as the source, or backend in the different node than the source) works and the other fails, manifesting as flakiness.

**Comment by nebril** [comment]
Hit here: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/5149/testReport/junit/Suite-k8s-1/18/K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy/

**Comment by jrajahalme** [comment]
Fixed by #12248
